homeassistant/helpers/template.py

```diff
@@ -1201,7 +1201,24 @@ def utcnow(hass: HomeAssistant) -> datetime:
     return dt_util.utcnow()
 
 
-def forgiving_round(value, precision=0, method="common"):
+def warn_no_default(function, value, default):
+    """Log warning if no default is specified."""
+    template = template_cv.get() or ""
+    _LOGGER.warning(
+        (
+            "Template warning: '%s' got illegal input '%s' when rendering template '%s' "
+            "but no default was specified. '%s' will return '%s', the template will fail "
```

Comment:
We should better clarify what is current behavior vs future behavior.

```suggestion
            "but no default was specified. Currently '%s' will return '%s', however this template will fail "
```